status/projects.py

```diff
@@ -477,8 +505,17 @@ def list_projects(self, filter_projects="all"):
                         final_projects[proj_id].get(end_date),
                     )
 
-            if len(projects_with_agreements[proj_id]) > 0:
-                final_projects[proj_id]["has_agreements"] = True
+        # Create slices of 200 projects each to get the agreements for
```

Comment:
Why slices of 200? Does it take too much time otherwise?

Reply:
Good question. I think it's fairly arbitrary. 200 seems to work. Larger number could become too large return body and a smaller number results in more requests.